source/deploy-strategies.rst

```diff
@@ -0,0 +1,162 @@
+
+Two projects use FabulAWS that I'm aware of: CCSR and OpenDebates. Both
+can experience heavy loads at times, with significant load changes over
+time.
+
+These days,
+`Elastic Beanstalk <https://aws.amazon.com/elasticbeanstalk/>`_
+is a credible alternative to FabuLAWS.
+It also supports autoscaling, using the same underlying AWS mechanisms
+as FabulAWS, and we're not responsible for maintaining
+it ourselves.
+
+Both FabulAWS and Elastic Beanstalk are tied to AWS.
+
+Margarita
+---------
+
+Most of our current client projects use a Caktus project called
+`Margarita <http://caktus.github.io/developer-documentation/margarita/margarita.html>`_
+for deployment, which uses the
+`Salt <https://saltstack.com/>`_ configuration management system
+to deploy the site.
+
+This has worked reasonably well (well enough that we keep using it), but
+over time we've discovered a number of shortcomings:
+
+* Difficulty debugging deploy problems
+* Fragility of Salt updates - many new updates seem to introduce new
+  bugs, and the next update that fixes those might have its own new
+  bugs. It can be tricky figuring out a version of Salt that works.
+* Resource usage - the Salt master process seems to grow over time to
+  use a lot of memory on the server, even though it shouldn't be doing
+  anything at all when we're not deploying. This ties up memory our
+  site could be using.
+
+We are therefore moving away from Margarita for new projects.
+
+Tequila
+-------
+
+Tequila (see `tequila-*` projects on Caktus github)
+is a successor project to Margarita, which uses
+`Ansible <http://docs.ansible.com/>`_
+instead of Salt. It is hoped that Tequila will avoid some of the problems
+of Margarita.
+
+* Ansible error messages are much clearer, making debugging deploy problems
+  easier.
+* Ansible is agent-less, so when not deploying, it takes no resources on
+  the servers.
+* Ansible seems (so far) to have a better track record of not introducing
+  regressions with its updates.
+
+At present, Tequila is in a beta state, with several newer projects trying
+it out.
+
+Elastic Beanstalk
+-----------------
+
+`Elastic Beanstalk <https://aws.amazon.com/elasticbeanstalk/>`_
+is an AWS service that provides a nice front-end wrapper
+around some of their other services, including Elastic Load balancer and
+autoscaling. A `Caktus blog post <https://www.caktusgroup.com/blog/2017/03/23/hosting-django-sites-amazon-elastic-beanstalk/>`_ gave a quick overview.
+
+An advantage of EB over Margarita, Tequila, and FabulAWS is that it provides
+a higher level of abstraction, so developers don't have to concern themselves
+with as many details about the deploy when they just want something that works.
+
+We inherited one project that was already using Elastic Beanstalk and have
+had good experiences with it. It was a big enough project that we were grateful
+for the horizontal scaling (multiple web servers behind a load balancer), though
+I don't think we have needed autoscaling.
```

Comment:
Actually BF scales up each afternoon and down late at night (from 2 to 4-5 servers) based on CPU usage